barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/ultra_circuit_builder.cpp

```diff
@@ -47,8 +47,10 @@ void UltraCircuitBuilder_<ExecutionTrace>::finalize_circuit(const bool ensure_no
             add_gates_to_ensure_all_polys_are_non_zero();
         }
         process_non_native_field_multiplications();
+#ifndef ULTRA_FUZZ
         process_ROM_arrays();
         process_RAM_arrays();
+#endif
         process_range_lists();
```

Comment:
Why are you processing range lists? They will add a lot of extra gates for small circuits